rtgui/histogrampanel.cc

```diff
@@ -869,105 +869,72 @@ void HistogramArea::updateBackBuffer ()
         LUTu& gh = rawMode ? ghistRaw : ghist;
         LUTu& bh = rawMode ? bhistRaw : bhist;
 
-        // make double copies of LUT, one for faster access, another one to scale down the raw histos
-        LUTu rhchanged(256), ghchanged(256), bhchanged(256);
-        unsigned int lhisttemp[256] ALIGNED16 {0}, chisttemp[256] ALIGNED16 {0}, rhtemp[256] ALIGNED16 {0}, ghtemp[256] ALIGNED16 {0}, bhtemp[256] ALIGNED16 {0};
-        const int scale = (rawMode ? 8 : 1);
-
-        for(int i = 0; i < 256; i++) {
-            if(needLuma) {
-                lhisttemp[i] = lhist[i];
-            }
-
-            if(needChroma) {
-                chisttemp[i] = chist[i];
-            }
-
-            if(needRed) {
-                rhchanged[i] = rhtemp[i] = rh[i] / scale;
-            }
-
-            if(needGreen) {
-                ghchanged[i] = ghtemp[i] = gh[i] / scale;
-            }
-
-            if(needBlue) {
-                bhchanged[i] = bhtemp[i] = bh[i] / scale;
-            }
-        }
-
         // Compute the highest point of the histogram for scaling
-        // Values at far left and right end (0 and 255) are handled differently
-
-        unsigned int histheight = 0;
+        unsigned int rgbmax = 0;
+        unsigned int lumamax = 0;
+        unsigned int chromamax = 0;
 
-        for (int i = 1; i < 255; i++) {
-            if (needLuma && lhisttemp[i] > histheight) {
-                histheight = lhisttemp[i];
+        for (int i = 1; i < 65535; i++) { // Values at far left and right end are handled differently
```

Comment:
@heckflosse This loop increased in size quite by a factor of 255, which might be the slowdown.